### PR TITLE
add README.md in hack/

### PR DESCRIPTION
**Pull request by tanjunchen**
**What type of PR is this?**

/kind documentation
 
**What this PR does / why we need it**:

I will update the document according to the views from everybody.

**Which issue(s) this PR fixes**:

ref:https://github.com/kubernetes/kubernetes/issues/86597

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

**Comment by hase1128** [comment]
@tanjunchen 
thanks creating the draft of readme.md.
I think it is better that all scripts are categorized according to purpose and use.

btw, I am not allowed to review because I'm not a member.
This seems a bit inconvenient, so I will check the requirements for becoming a member.

**Comment by tanjunchen** [comment]
@hase1128  
yes . the title contains WIP (work in progess) , I want to know others  opinions from kk.

**Comment by tanjunchen** [comment]
/assign @fejta 
@hase1128
could you help me review this pr ? I am not familiar with some scripts, so I have not added any related introductions. If everybody has any idea or docs about them , please comment.

File: hack/README.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+and gives a brief introduction and explanation of these scripts. 
+
+## Overview
+
+The [`hack`](.) directory contains many scripts that ensure continuous development of kubernetes, 
+enhance the robustness of the code, improve development efficiency, etc. 
+The explanations and descriptions of these scripts are helpful for contributors. 
+For details, refer to the following guidelines.
+
+## Key scripts
```
Comment (fejta) [inline hack/README.md]: This seems to just list every file, can you pick "key" scripts?

How about if we just tell people to run `verify-all.sh` before submitting a PR and if anything fails run `update-all.sh`?

Anyone interested in more detail should probably just look at the scripts themselves.

Listing every script this way is going to be out of date almost as soon as its submitted, and manually making a list of every file doesn't do anything particularly different from what github already does.

File: hack/README.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#conformance-test-requirements. 
+* [`hack/e2e-internal`](e2e-internal): 
+* [`hack/gen-swagger-doc`](gen-swagger-doc): This script generates docs from the latest swagger spec. 
+* [`hack/jenkins`](jenkins): Jenkins is no longer in use, please see the [test-infra repo](https://github.com/kubernetes/test-infra) 
+and in particular [Prow](https://github.com/kubernetes/test-infra/tree/master/prow) instead.
+* [`hack/lib`](lib): This directory contains some common scripts, such as `etcd.sh`, `golang.sh`, `init.sh`, `logging.sh`, `protoc.sh`, `swagger.sh`, `test.sh`, `util.sh`, `version.sh`. 
+* [`hack/make-rules`](make-rules): 
+* [`hack/testdata`](testdata): The directory contains many test data, such as yaml files, json files, tls.crt, tls.key, etc. 
+* [`hack/verify-flags`](verify-flags): This `excluded-flags.txt` file contains some flags used by `hack/verify-flags-underscore.py`. 
+* [`.golint_failures`](.golint_failures): This file contains the result after running `hack/verify-golint.sh`.
```
Comment (fejta) [inline hack/README.md]: This seems like something that should go in verify-golint.sh, same for the other helper files

File: hack/README.md (hunk 1 above)
Comment (hase1128) [reply hack/README.md]: I think it's a good idea to pick up only key scripts and add guidance for each category(verify, update and etc.) in readme.md. 
Are `verify-all.sh`  and `update-all.sh` the only key scripts?

Comment (fejta) [reply hack/README.md]: Yes, as they run all the verify and update scripts respectively.

Comment (fejta) [reply hack/README.md]: I still think this could separate out update-all and verify-all into their own primary section

**Review by fejta — COMMENTED** [review]
/lgtm
/approve

File: hack/README.md (hunk 1 above)
Comment (tanjunchen) [reply hack/README.md]: sure i will update it

**Pull request merged**